src/v/compat/commit_tx_compat.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cluster/types.h"
+#include "compat/check.h"
+#include "compat/commit_tx_generator.h"
+#include "compat/json.h"
+
+namespace compat {
+
+/*
+ * cluster::commit_tx_request
+ */
+template<>
+struct compat_check<cluster::commit_tx_request> {
```

Comment:
Maybe you can cherry pick my macro which has been moved into `compat/check.h` by Noah in his PR here https://github.com/redpanda-data/redpanda/pull/5831/commits/ce557a1bc95076422e031d3751bce46dc1ab9595

Reply:
I'd prefer leaving this PR dependency-free. Maybe in the next round of types

Reply:
KLOCs is life

Reply:
Noah's PR is merged, so I'll pick this up.

Reply:
Done